pkg/web/import_blockchain_data.go

```diff
@@ -4,32 +4,25 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"net/http"
-	"sync"
+	"sync/atomic"
 
 	dogeboxd "github.com/dogeorg/dogeboxd/pkg"
 )
 
 var (
-	importInProgress bool
-	importMutex      sync.Mutex
+	importInProgress int32 // 0 = false, 1 = true
```

Comment:
Can’t we keep this as a book?

There’s a [ComparendSwap method on the type](https://pkg.go.dev/sync/atomic#Bool.CompareAndSwap)